beacon-chain/rpc/eth/events/events.go

```diff
@@ -164,13 +164,22 @@ func (s *Server) StreamEvents(w http.ResponseWriter, r *http.Request) {
 		httputil.HandleError(w, msg, http.StatusInternalServerError)
 		return
 	}
-	es, err := newEventStreamer(eventFeedDepth, s.KeepAliveInterval)
+	depth := s.EventFeedDepth
+	if depth == 0 {
+		depth = DefaultEventFeedDepth
+	}
+	es, err := newEventStreamer(depth, s.KeepAliveInterval)
 	if err != nil {
 		httputil.HandleError(w, err.Error(), http.StatusInternalServerError)
 		return
 	}
-	if err := es.streamEvents(ctx, sw, topics, s); err != nil {
-		log.WithError(err).Debug("Event streamer shutting down due to error.")
+
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+	api.SetSSEHeaders(w)
```

Comment:
seems like it probably doesn't make a difference but maybe using sw will look more consistent.